pycozmo/behavior.py

```diff
@@ -44,21 +177,75 @@ def __init__(self,
                  behavior_id: str,
                  needs_action_id: str,
                  display_name_key,
-                 params: Dict[str, Any]) -> None:
+                 params: BehaviorParameters) -> None:
         self.behavior_class = str(behavior_class)
         self.id = str(behavior_id)
         self.needs_action_id = str(needs_action_id)
         self.display_name_key = str(display_name_key)
-        self.params = dict(params)
+        self.params = params
+
+    @classmethod
+    def from_json(cls, data: Dict):
+        return cls(behavior_class=data['behaviorClass'],
+                   behavior_id=data['behaviorID'],
+                   needs_action_id=data.get('needsActionID', ''),
+                   display_name_key=data.get('displayNameKey', ''),
+                   params=BehaviorParameters.from_json(data.get('params')) if 'params' in data else None)
 
 
-def load_behaviors() -> Dict[str, Behavior]:
-    # TODO: Load cozmo_resources/config/engine/behaviorSystem/behaviors/*/*.json
+class TestBehavior:
+    def __init__(self,
+                 behavior_class: str,
+                 behavior_id: str,
+                 loop_forever: Optional[bool] = False,
+                 test_gap: Optional[float] = 0.0,
+                 runs_per_test: Optional[int] = 1) -> None:
```

Comment:
Is `Optional` necessary? Are there cases in which `loop_forever`/`test_gap`/`runs_per_test` are `None`?

Reply:
Actually only `DevTurnInPlaceTest` has these parameters. The rest of the tests only include the behaviour class and id.
This is not a very critical class, since there are only a few tests in the jsons, and maybe will never be used. I included it only to differentiate the tests from the actual behaviours.

Another note on this, something that maybe I should change: I have added the Optional tag to all parameters that do not appear every time on the configuration files. This might not be necessary since the json loading already takes care of assigning a value to the class object, but I thought it would make more clear that in some cases these parameters are not appearing in the configs.

Reply:
Let's keep it as is then - makes sense.